File: LaborMarket.Api/Models/UserModel.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LaborMarket.Api.Models
{
	public class UserModel
	{
		[Key]
		[JsonIgnore]
		public int UserId { get; set; }
		public string FirstName { get; set; } = null!;
		public string LastName { get; set; } = null!;
		public string Email { get; set; } = null!;
		public string PhoneNumber { get; set; } = null!;
		public string PasswordHash { get; set; } = null!;
		public DateTime CreatedAt { get; set; }
		public string? CV { get; set; }
	}
}

File: LaborMarket.Web/src/app/models/user-model.ts
export type UserDataModel = {
  userId: number;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  passwordHash: string;
  createdAt: string;
  cv: string;
  profileImageUrl?: string;
};

export type CreateUserModel = {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  password: string;
  createdAt: string;
  role: string;
};
